### PR TITLE
New Mime Ability: Invisible Touch

### PR DESCRIPTION
**Pull request by Super3222**
#### Intent of your Pull Request

Mimes can now make objects and structures vanish right before your eyes using the "Invisible Touch".
This PR adds an ability - which operates much like ei nath or stone spells - that can be used to turn things invisible. Cast the spell, a touch attack object will appear in your hand. All you need to do is attack something within close proximity, and it'll turn invisible for a few seconds, and than reappear. You get at least 3 uses before the invisible touch deletes itself. 

Mimes can toggle the visibility of chairs !!!INDEFINITELY!!!! So they look like gods when the shuttle is taking off.

The Invisible Touch can also toggle the mimes invisibility. However this wastes all of their uses, and makes them flicker on and off for at least 5-6 seconds.

The Invisible Touch has limits. For example none of the following can be messed with -
- Bomb cores
- Cherry Bombs
- Grenades
- The Self Destruct Terminal
- All Guns in general

I also tweaked the talking ability. Mimes cannot break their vow until the round's been 25 minutes in.

##### Changelog

:cl:
rscadd: The Mime can now turn things invisible! The Invisible Touch, as they call it, can turn objects and structures invisible for a certain amount of time.
rscadd: Mimes can also use this ability on themselves for a flickering effect that won't last long.
tweak: The round has to be 25 minutes in until the mime can break their vow.
/:cl:

### CODE REVIEW

File: code/modules/spells/spell_types/godhand.dm [hunk 1]
```diff
@@ -131,6 +131,9 @@
 	if(iscarbon(target))
 		if(target == user)
 			if(user.job == "Mime")
+				if(uses < 3)
+					user << "<span class='warning'>You've got to have more chargers than that!</span>"
```
Comment (Cruix) [inline code/modules/spells/spell_types/godhand.dm]: >chargers
>not using superior chaos/lightning holo

**Comment by ghost** [comment]
Don't like this. Mime is supposed to use invisible objects, not make bombs invisible and throw them while nobody can see.

Remember that 3 seconds is the prime-time for a screwdrivered chemgrenade. IT's not hard to just make an invisible fucking grenade and toss it when nobody can see.

I'd prefer the mime being able to close invisible doors and firelocks on people or something tbqh

**Comment by Super3222** [comment]
Ah, bombs. I should've considered them as well. I'll make a blacklist for things the mime can't turn invisible. Bombs included.

File: code/modules/spells/spell_types/godhand.dm [hunk 2]
```diff
@@ -110,3 +110,79 @@
...
+		if((target in things))
+			user << "<span class='warning'>You can't use this on the same thing more than once!</span>"
+			return
+		if(!target.alpha)
+			return
+		things += target
+		user << "<span class='warning'>You poke [target] extinguishing one of your charges.</span>"
+		uses--
+		target.alpha = 0
+		spawn(150)
```
Comment (Cruix) [inline code/modules/spells/spell_types/godhand.dm]: 15 seconds is too long. Also spawn is banned, use addtimer().

Comment (Super3222) [reply code/modules/spells/spell_types/godhand.dm]: What? Coding standards on Yogstation? I guess we're finally moving forward.

**Pull request merged**